src/main/daml/Daml/Finance/Util/Date/DayCount.daml

```diff
@@ -4,21 +4,101 @@
 module Daml.Finance.Util.Date.DayCount
   ( calcDcf
   , calcDcf30E360
+  , calcPeriodDcf
+  , calcPeriodDcfActActIsda
+  , calcPeriodDcfActActIsma
   ) where
 
 import DA.Date
-import Daml.Finance.Interface.Types.Date.DayCount(DayCountConventionEnum(..))
+import Daml.Finance.Interface.Types.Date.DayCount
+import Daml.Finance.Interface.Types.Date.RollConvention (Period(..), PeriodEnum(..))
+import Daml.Finance.Interface.Types.Date.Schedule (Frequency, SchedulePeriod, StubPeriodTypeEnum(..))
+import Daml.Finance.Util.Date.RollConvention (next, previous)
 
 -- | Calculates the day count fraction given the correponding convention.
 -- Currently 30E360 is not supported as we do not want to expose the maturity date of the product as an additional parameter.
 calcDcf : DayCountConventionEnum -> Date -> Date -> Decimal
 calcDcf _ f t | f > t = error "fromDate is greater than toDate"
 calcDcf Act360 f t = calcDcfAct360 f t
 calcDcf Act365_Fixed f t = calcDcfAct365Fixed f t
+calcDcf ActAct_ISDA f t = error "This calculation requires the knowledge of the stub period. Please, call `calcPeriodDcf` instead."
+calcDcf ActAct_ICMA f t = error "This calculation requires the knowledge of the stub period. Please, call `calcPeriodDcf` instead."
 calcDcf Basis_30360 f t = calcDcf30360 f t
 calcDcf Basis_30360_ICMA f t = calcDcf30360Icma f t
 calcDcf Basis_30E3360 f t = calcDcf30E360 False f t -- this corresponds to 30E/360 without the maturity date adjustment
-calcDcf Basis_30E360 f t = error "This calculation requires the knowledge of the maturity date of the product. Please, use `calcDcf30E360 instead`."
+calcDcf Basis_30E360 f t = error "This calculation requires the knowledge of the maturity date of the product. Please, use `calcDcf30E360` instead."
+
+-- | Calculate day count fraction for a schedule period.
+-- It takes the following parameters:
+--   DayCountConventionEnum: to specify which day count convention should be used
```

Comment:
using a bullet point e.g. 

```suggestion
--  - DayCountConventionEnum: to specify which day count convention should be used
```

might make the doc look nicer, should be tested though as I'm not entirely sure